File: recursion/java/Recursion4.java
import java.util.Scanner;

public class Recursion4 {

    public static void printFirstNEvenNaturalNumberReverse(int n) {
        // negative number
        if (n <= 1) {

            return;
        }

        if (n == 2) {
            System.out.print(n + ", ");
            return;
        }
        // make even
        if ((n & 1) >= 1) {
            n = n - 1;
        }

        System.out.print(n + ", ");
        printFirstNEvenNaturalNumberReverse(n - 2);

    }

    public static Scanner scan = new Scanner(System.in);

    public static void main(String... args) {
        int n = scan.nextInt();
        printFirstNEvenNaturalNumberReverse(n);
    }
}

File: recursion/go/Recursion4.go
package main

import "fmt"

// first n natural number
func printFirstNEvenNaturalNumberReverse(number int) {
	// negative number
	if number <= 1 {
		return
	}

	// base case
	if number == 2 {
		fmt.Print(2, ", ")
		return
	}

	// start with even 
if(number & 1 >= 1){
	 number = number -1
}
	// recursive call
	fmt.Print(number, ", ")
	printFirstNEvenNaturalNumberReverse(number - 2)

}
func main() {
	var number int
	fmt.Scan(&number)
	printFirstNEvenNaturalNumberReverse(number)
}
